lib/puppet/provider/docker_compose/ruby.rb

```diff
@@ -14,8 +14,7 @@
   end
 
   has_command(:docker_compose, command(:dockercompose)) do
-    Dir.mkdir('/tmp_docker') unless Dir.exist?('/tmp_docker')
-    ENV.store('TMPDIR', '/tmp_docker')
+    environment(HOME: '/root')
```

Comment:
Overriding $HOME unconditionally is probably a bad idea. This is still a draft so probably needed for some tests, but flag it so that we don't forget about it by mistake.

Reply:
This seems to be true for other command definitions. It's a temporary override but I do see your point.

Reply:
Please please please do not create new directories in /.
Please pick some other place (for example under /usr/local/share, /opt, /var/cache, or anywhere else but /).

Reply:
Hey @bjvrielink, I think you may have misinterpreted the proposed change here. 

This PR removes the directory creation.

Reply:
Yeah, now I noticed. I've spent a little too much time right now trying to figure out what was/is broken in the current release.

Reply:
We are working on a fix 🙂

For now, I would pin to the previous version.

Reply:
I did some runtime tests with the new changes, and line 27 of lib/puppet/provider/docker_compose/ruby.rb was failing as it wasn;t getting the new tmpdir param.

I got it to work by changing line 27 of lib/puppet/provider/docker_compose/ruby.rb to
```
compose_output = YAML.safe_load(exec_dockercompose(args))
```
however, it is not idempotent as the TMPDIR var gets set every time the resource runs.  I can't find a way to silence it.

Reply:
Good spot! 

Interesting regarding the idempotency. I wouldn't have expected it to have an impact.

It's also highly possible that my suggested implementation is not the right fit here.

Maybe we shouldn't be overridden TMPDIR at all given that it it isn't actually a docker setting.

Could this boil down to a simple documentation update where we advise that impacted users may want to override TMPDIR with another puppet resource?

@smortex @kenyon what are your thoughts?

Reply:
I tried setting the TMPDIR env var previously using another puppet resource.  However without a reboot the docker_compose resource wasn't seeing/respecting it.  I like your draft approach as it takes care of it close to the compose resource, it just needs to occur silently if the param is declared - interestingly the existing (breaking) approach did that, however my testing with a hybrid method wasn;t reading the resource[:tmpdir] if set earlier.

Reply:
An alternative approach may be to precede the docker-compose command with 'TMPDIR=resource[:tmpdir]' so that it occurs as part of the command if the tmpdir param is declared.  That way ruby isn't doing it, the shell is.  Just a thought.